OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/CourseModalDialogFragment.kt

```diff
@@ -187,7 +187,7 @@ class CourseModalDialogFragment : DialogFragment() {
             retryListener = DialogInterface.OnClickListener { _, _ ->
                 // already purchased course.
                 iapViewModel.iapFlowData.isVerificationPending = false
-                iapViewModel.iapFlowData.flowType = IAPFlowData.IAPFlowType.USER_INITIATED
+                iapViewModel.iapFlowData.flowType = IAPFlowData.IAPFlowType.SILENT
```

Comment:
better to verify it with iOS team.

Reply:
this is the use case when the course is already purchased, and the user just needs to refresh the course content. in this case, the app needs to execute the flow.
- Show the full-screen loader.
- Refresh the course data.
- Hide the full-screen loader.
- User can view the locked content.